simplified-opds2-r2/src/main/java/org/nypl/simplified/opds2/r2/internal/OPDS2ParserR2.kt

```diff
@@ -0,0 +1,51 @@
+package org.nypl.simplified.opds2.r2.internal
+
+import org.nypl.simplified.opds2.OPDS2Feed
+import org.nypl.simplified.parser.api.ParseError
+import org.nypl.simplified.parser.api.ParseResult
+import org.nypl.simplified.parser.api.ParserType
+import org.readium.r2.opds.OPDS2Parser
+import java.io.InputStream
+import java.net.URI
+
+class OPDS2ParserR2(
+  private val uri: URI,
+  private val stream: InputStream,
+  private val warningsAsErrors: Boolean
+) : ParserType<OPDS2Feed> {
+
+  override fun parse(): ParseResult<OPDS2Feed> {
+    return try {
+      val data = OPDS2Parser.parse(stream.readBytes(), uri.toURL())
+      ParseResult.Failure(
+        warnings = listOf(),
+        errors = listOf(
+          ParseError(
+            source = this.uri,
+            message = "Not implemented yet!",
+            line = 0,
```

Comment:
I've left this parser in place in order to keep our options open with regards to OPDS 2.0 parsers. I'd like to finish this at a slightly later date and do a competitive analysis between the R2 and Irradia parsers. Right now, we can't use R2 for the accounts registry as it doesn't support the custom roles and other things the accounts registry feeds use.